feature_tests/steps/result_steps.py

```diff
@@ -0,0 +1,87 @@
+from behave import then
+from domain.core.reference import Reference
+
+from feature_tests.steps.common import parse_value, read_value_from_path
+
+
+@then("the result is of type {type_name}")
+def step_impl(context, type_name):
+    t = type(context.result).__name__
+    assert t == type_name, f"Expected '{type_name}' got '{t}'"
+
+
+@then("the result {field} equals {value:Number}")
+def step_impl(context, field, value):
+    assert context.result is not None, "result"
+    v = context.result.__dict__[field]
+    assert v == value, f"Expected {value} but found {v}"
+
+
+@then('the result {field:String} equals "{value:String}"')
+def step_impl(context, field, value):
+    assert context.result is not None, "result"
+    v = context.result.__dict__[field]
+    assert v == value, f"Expected {value} but found {v}"
+
+
+@then("the result {field:String} equals {value:UUID}")
+def step_impl(context, field, value):
+    assert context.result is not None, "result"
+    v = context.result.__dict__[field]
+    assert v == value, f"Expected {value} but found {v}"
+
+
+@then('the result {field:Path} is a Reference to {system:String} "{value:String}"')
+def step_impl(context, field, system, value):
+    ref = context.result.__dict__[field]
+    assert isinstance(ref, Reference), "Is reference"
+    assert ref.system == system, f"Expected '{system}' got '{ref.system}'"
+    assert ref.value == value, f"Expected '{value}' got '{ref.value}'"
+
+
+def _assert_value(obj, path, value):
```

Comment:
should probably be calling the other similar function for accessing path objects, to avoid duplication. Otherwise, same comment regarding a docstring